Update documentation (#6)
MiaAltieri/mongodb-operator: CONTRIBUTING.md

# mongodb-operator

## Developing

> As this is a machine charm, it can be deployed locally using [Juju on a local LXD cloud](https://juju.is/docs/olm/lxd). Once the controller is created, to deploy this workload you need to follow some preliminar steps.

Reviewer: Definitely, I **loved** that you did this, I wanted to change it from the Passive Voice to the Active Voice, since I think it makes things more clear. This is a nit :)

> 

Reviewer: If you do not already have a Juju controller bootstrapped to an LXD `localhost` cloud you can set one up by doing the following:

```bash
# install requirements 
sudo snap install charmcraft --classic
sudo snap install lxd
sudo snap install juju --classic

# configure lxd
sudo adduser $USER lxd
newgrp lxd
lxd init --auto
lxc network set lxdbr0 ipv6.address none

# boostrap controller to lxd
juju clouds
juju bootstrap localhost overlord
```

> Afer downloading the repository, create and activate a virtualenv, and install the development requirements:

Reviewer: ```suggestion
After cloning the repository, create and activate a virtualenv, and install the development requirements:
```

```bash
$ virtualenv -p python3 venv
$ source venv/bin/activate
$ pip install -r requirements.txt
```

> ### Build

Reviewer: ```suggestion
### Building the Charm
```

> ```bash
> $ charmcraft pack
> ```

Reviewer: ```suggestion
```bash
cd mongodb-operator
charmcraft pack
```
```

> ### Deploy

Reviewer: ```suggestion
### Deploying the Charm
```

```bash

> $ juju deploy ./mongodb_ubuntu-20.04-amd64.charm

Reviewer: remove `$` it makes it difficult for people to copy and paste into the terminal :-)

tox                  # runs 'lint' and 'unit' environments
```

## Code overview

TODO: expand on this section when spec doc is created

## Intended use case

TODO: expand on this section when spec doc is created

## Roadmap

> TODO: expand on this section when spec doc is created

Reviewer: Add the features we plan on adding that we don't already have, ie replication, sharding, backups, security :)

## Canonical Contributor Agreement

> https://ubuntu.com/legal/contributors

Reviewer 2: ```suggestion
Canonical welcomes contributions to the Charmed MongoDB Operator. Please check out our [contributor agreement](https://ubuntu.com/legal/contributors) if you're interested in contributing to the distribution.
```
